src/workerd/util/sqlite.c++

```diff
@@ -174,16 +181,24 @@ class SqliteCallScope {
     /* SQLITE_MISUSE doesn't put error info on the database object, so check it separately */      \
     KJ_ASSERT(_ec != SQLITE_MISUSE, "SQLite misused: " #code, ##__VA_ARGS__);                      \
     if (_ec == SQLITE_IOERR) sqliteCallScope.rethrowVfsError();                                    \
+    if (isCriticalError(code)) {                                                                   \
+      regulator.onCriticalError(_ec, dbErrorMessage(code, db));                                    \
```

Comment:
We need to actually verify whether a transaction was rolled back.
- There might not have been a transaction open. In this case `db.savepoints` will be empty.
- If there was a transaction open, SQLite did not necessarily roll it back. [Per the docs](https://www.sqlite.org/lang_transaction.html#response_to_errors_within_a_transaction), we need to call `sqlite3_get_autocommit()` to find out.

We don't want to do all this in a macro, of course, so we probably want a method like `db.checkForCriticalError(code)`.